libexec/bootstrap-scripts/main-deffile.sh

```diff
@@ -50,7 +50,7 @@ if [ -n "${BOOTSTRAP:-}" -a -z "${SINGULARITY_BUILDNOBASE:-}" ]; then
 fi
 
 # take a snapshot of the environment for later comparison
-if [ ${BOOTSTRAP:-} = "localimage" ]; then
+if [ ${BOOTSTRAP:-} = "localimage" -o ${BOOTSTRAP:-} = "shub" ]; then
     SINGULARITY_STARTING_ENVIRONMENT=$(eval_abort env -i ${SINGULARITY_libexecdir}/singularity/helpers/record-env.sh ${SINGULARITY_ROOTFS})
```

Comment:
ok cool, so this deals with environment / labels in the same way as a localimage?

Reply:
Yepper!  I think I might be able to add another `-o docker` and make it work in the same way for docker too.  But I haven't tried it yet.